src/System.Memory/src/System/Buffers/Text/Utf8Formatter/Utf8Formatter.Integer.Signed.Default.cs

```diff
@@ -2,47 +2,166 @@
+                }
+                else
+                {
+                    return TryFormatInt64MoreThanQuintillion(value, buffer, out bytesWritten);
+                }
+            }
+#else
+            return TryFormatInt64MultipleDigits(value, buffer, out bytesWritten);
+#endif // BIT32
+        }
+
+        // TODO: Use this instead of TryFormatInt64Default to format numbers less than int.MaxValue for BIT32
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool TryFormatInt32Default(int value, Span<byte> buffer, out int bytesWritten)
+        {
+            if ((uint)value < 10)
+            {
+                return TryFormatUInt32SingleDigit((uint)value, buffer, out bytesWritten);
+            }
+            return TryFormatInt32MultipleDigits(value, buffer, out bytesWritten);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool TryFormatInt32MultipleDigits(int value, Span<byte> buffer, out int bytesWritten)
+        {
+            if (value < 0)
+            {
+                value = -value;
+                int digitCount = FormattingHelpers.CountDigits((uint)value);
+                // WriteDigits does not do bounds checks
+                if (digitCount >= buffer.Length)
+                {
+                    bytesWritten = 0;
+                    return false;
+                }
+                buffer[0] = Utf8Constants.Minus;
+                FormattingHelpers.WriteDigits((uint)value, buffer.Slice(1, digitCount));
+                bytesWritten = digitCount + 1;
                 return true;
             }
-            
+            else
+            {
+                return TryFormatUInt32MultipleDigits((uint)value, buffer, out bytesWritten);
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool TryFormatInt64MultipleDigits(long value, Span<byte> buffer, out int bytesWritten)
+        {
             if (value < 0)
             {
                 value = -value;
                 int digitCount = FormattingHelpers.CountDigits((ulong)value);
-                if (digitCount >= buffer.Length) goto FalseExit;
+                // WriteDigits does not do bounds checks
+                if (digitCount >= buffer.Length)
+                {
+                    bytesWritten = 0;
+                    return false;
+                }
+                buffer[0] = Utf8Constants.Minus;
+                FormattingHelpers.WriteDigits((ulong)value, buffer.Slice(1, digitCount));
                 bytesWritten = digitCount + 1;
+                return true;
+            }
+            else
+            {
+                return TryFormatUInt64MultipleDigits((ulong)value, buffer, out bytesWritten);
+            }
+        }
+
+        // Split long into two parts that can each fit in a uint - {1-9 digits}{9 digits}
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
I think that ForceInline on methods that are this big is going to hurt on average, especially the small numbers.